src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/ExecutorFactory.cs

```diff
@@ -11,7 +11,7 @@ namespace Microsoft.AspNetCore.Mvc.RazorPages.Internal
 {
     public static class ExecutorFactory
     {
-        public static Func<Page, object, Task<IActionResult>> CreateExecutor(
+        public static Func<Page, object, object[], Task<IActionResult>> CreateExecutor(
```

Comment:
Should we change the signature of the type to `Func<object, object[], Task>`? We needed the `Page` in the past because the binder was a property on it. We can have the invoker determine what to pass in now and reduce the places of how the handler is discovered.